### PR TITLE
fix: use auto-imports to enable global `$fetch`

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

There are currently ssues with using `vercel-edge` and other presets where we are adding globals into `globalThis` - e.g. https://github.com/pi0/nuxt-on-the-edge/blob/main/app.vue#L11. This PR is a starting point for discussion.

I propose that if we go down this route, we add a warning if user explicitly disables nitro auto-imports, stating that they will need to import `$fetch` explicitly in that case.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
Can we please keep global injection behavior _in addition_ to auto imports?

**Comment by pi0** [comment]
globalThis issue with vercel's edge is so strange indeed. Can you please point out "and other presets" currently suffering this issue? 

Regardless even if going down this path, i think we can keep hybrid approach to both inject and auto import.

**Comment by danielroe** [comment]
Personally I only experienced the issue with vercel-edge but I imagined it might be replicable on netlify-edge or cloudflare.

File: src/types/fetch.ts [hunk 1]
```diff
@@ -93,15 +93,4 @@ export interface $Fetch<
   ): $Fetch<T, R>;
 }
 
-declare global {
```
Comment (pi0) [inline src/types/fetch.ts:96]: Can we please still keep this? I like to make types working as much as possible before prepare command.

Comment (danielroe) [reply src/types/fetch.ts:96]: I am worried that the types may conflict but I'm happy to restore it and see.

**Comment by harlan-zw** [comment]
This would be nice, vercel-edge does throw the `undefined $fetch`. If you import directly from `ofetch` it breaks the internal route handling.

Workaround was adding `globalThis.` to any requests

```ts
await globalThis.$fetch(path)
```

**Comment by pi0** [comment]
I finally got a chance to make a minimal reproduction. Seems `globalThis` support issues are really limited to `vercel-edge` preset and other workers are ok. Wile using auto imports, might handle edge case of `$fetch` on vercel-deploy, there are many more cases we are already using globalThis to polyfill and such workaround cannot work for them.

Let's track issue in https://github.com/unjs/nitro/issues/921. 

(as a better workaround maybe we can only for `vercel-edge` transform `$fetch` to `globalThis.` using rollup provide?)

**Pull request closed without merging**